### ISSUE TITLE
UDP and lan discovery greyed out by default

### ISSUE THREAD

**Issue by doncatnip** [qTox/qTox#5311]
##### Brief Description

OS: Linux, Debian 9
qTox version: v1.16.3.
Commit hash:  8eed684c37bd711de2ca4bd863f5f0509edcdc2d
toxcore:   0.2.3
Qt:  5.11.1
Hardware:  pretty standard

qTox was installed using flatpak io.github.qtox.qTox since Debian does not offer any other (obvious) way to do it anymore.

Reproducible: Always

##### Steps to reproduce

1. Install qTox via flatpak
2. Run qTox
3. check advanced config

##### Observed Behavior

Enable UDP is disabled and greyed out.
LAN discovery is disabled and greyed out.

##### Expected Behavior

Enable UDP is not greyed out.
LAN discovery is not greyed out.
And if they are greyed out, I would expect to see some info as to why (missing libs ?)

##### Additional Info

Unfortunately, I had to install qTox via flatpak. It is my only package ever installed via flatpak.
![tox_udp_greyedout](https://user-images.githubusercontent.com/692660/44726957-2f1f3680-aad9-11e8-8e56-2e05fda223cc.png)

**Comment by sudden6** [comment]
Did you select `Proxy type: None`? If not, then that's the reason they are greyed out. By default it should be `None` so this may be a UI glitch.

**Comment by doncatnip** [comment]
Well, that was easy - and a little bit embarrassing.
Proxytype was '' (empty string). I changed it to None and those settings are no longer greyed out.
Thank you.

**Comment by sudden6** [comment]
I guess this was with a completely fresh profile? If so I'll let this open, because "broken by default" is probably bad design :/

**Comment by doncatnip** [comment]
Actually, it was an imported profile. The original does have the same empty settings. I did not notice because I'm on a much more permissive network with that one. I didn't tweak those settings so maybe it happened during some version change or previous export/import.

Correction: this is indeed the default with new profiles on Arch linux, using the native package and on Debian w/ flatpak.

**Comment by sudden6** [comment]
> Correction: this is indeed the default with new profiles on Arch linux, using the native package and on Debian w/ flatpak.

Thanks, then we have to fix that...

**Comment by rootkea** [comment]
I just experienced this issue of Debian 10 with qTox v1.16.3-2debian. On fresh qTox install both "Enable UDP" and "Enable LAN Discovery" was greyed out with no way to know what caused that.

I thought the work was yet to be done and maybe I had the older version so even tried with flatpak but observed the same. And then a query on the Internet landed me here.

So this issue isn't fixed yet. I suggest selecting proxy type as "None" by default and at least not greying out both the UDP and LAN discovery options.

Please reopen this issue.

**Comment by anthonybilinski** [comment]
The fix was made after v1.16.3, so won't be in releases until v1.17.0. Current nightlies should both have "None" selected by default, and also heal saved invalid values to "None".

Which version of the flatpak were you using, @rootkea?

**Comment by rootkea** [comment]
@anthonybilinski 
```
$ flatpak --version
Flatpak 1.4.2
$ cat /etc/issue
Ubuntu 18.04 LTS
```
qTox version: 1.16.3
Commit hash: 8eed684c37bd711de2ca4bd863f5f0509edcdc2d
toxcore version: 0.2.3
